DOC-2653 Change Impact Analysis Quickstart - Harness feature flag as change source. (#788)
harness/developer-hub: docs/service-reliability-management/howtos-service-reliability-management/change-impact-analysis/change-impact-analysis-quickstart.md

---
title: Create a monitored service
description: Create a monitored servie to correlate change events.
sidebar_position: 10
helpdocs_topic_id: lwi37ku94x
helpdocs_category_id: 7lvetjzo1m
helpdocs_is_private: false
helpdocs_is_published: true
---

Harness Change Impact Analysis allows you to get a clear understanding of what has changed, when it changed, why it changed, and the effect it had on your monitored service.  Identifying recent change events in your service around the time when there is deterioration of performance helps in quickly and accurately identifying the root cause of the issue that led to the SLO violations.

These changes might include deployments, infrastructure changes, incidents, and feature flags. Harness Change Impact Analysis helps you: 

> - Correlate recent changes in your service and their broader impact on your service.

Reviewer: on your service's health

Author: Updated

 
To create a new change source, do the following:

1. Select **+ Add New Change Source**.  
   The Add New Change Source dialog appears.  

2. In the **Provider Type** field, select the change source for which you want to monitor the changes. Following are the available options:  

    - **Deployments**: You can choose either **Harness CD NextGen** or **Harness CD**. If you choose **Harness CD**, then in the **Connect Change Source** section, select the **Application** for which you want to monitor the changes.  
    - **Incident**: Select the incident management tool which monitors your application.

3. In the **Source Name** field, enter a name for the change source that you are configuring.  
The new change source appears in the **Change Sources** list.

### Select a health source

A health source is an APM or log provider using which the Harness Monitored Service collects the metrics and logs of your service. 

Select the health source which monitors the health of your service.

To create a new health source, do the following:  

> 1. Select **+ Add New Health Source**.  

Reviewer: It should be seperate section. Just say add a new health source.
Now, Health source section(module) can talk about how to configure each type of health source

Reviewer: same in health source. just say add a connection and divert the user to add connector which can have seperate doc for each type of connector.

Author: Noted. As discussed, these issues will be addressed while restructuring the documentation.

---
title: Create a monitored service
description: Create a monitored servie to correlate change events.
sidebar_position: 10
helpdocs_topic_id: lwi37ku94x
helpdocs_category_id: 7lvetjzo1m
helpdocs_is_private: false
helpdocs_is_published: true
---

> Harness Change Impact Analysis allows you to get a clear understanding of what has changed, when it changed, why it changed, and the effect it had on your monitored service. It helps you in identifying recent change events in your service around the time when there is deterioration of performance. This real time change event tracking aids in faster root cause analysis allowing teams to make confident decisions.

Author: Updated

> 

Reviewer 2: Harness Change Impact Analysis allows you to get a clear understanding of what has changed, when it changed, why it changed, and the effect it had on your monitored service. It helps you identify recent change events in your service around the time when there is a deterioration of performance. This real-time change event tracking aids in faster root cause analysis, allowing teams to make confident decisions.

Author: Updated text:

Harness Change Impact Analysis enables you to understand what has changed, when it changed, why it changed, and the effect it had on your monitored service. It helps you identify recent change events in your service around the time performance deteriorated. These changes might include deployments, infrastructure changes, incidents, and feature flags. This real-time change event tracking aids in faster root cause analysis and allows teams to make confident decisions.

> These changes might include deployments, infrastructure changes, incidents, and feature flags. Harness Change Impact Analysis helps you: 

Author: Makes sense! Moved the "These changes might include deployments, infrastructure changes, incidents, and feature flags." part to the first paragraph and removed the list that starts with "Harness Change Impact Analysis helps you: ".

- Correlate recent changes in your service and their broader impact on the health of your service.
- Triage the changes quickly and reduce downtime.

This topic describes how to create a monitored service to correlate change events.

## Prerequisites

> - Ensure that you have installed a Delegate and it can communicate with Harness, health source, and change source. To learn more about Delegates, go to [Get started with delegates](https://developer.harness.io/docs/category/get-started-with-delegates).

Reviewer 2: Ensure that you have installed a Delegate and that it can communicate with Harness, health source, and change source.

> - You must have created an Organization in Harness. To learn more about Organizations in Harness, go to [Create Organizations and Projects](https://developer.harness.io/docs/platform/organizations-and-projects/create-an-organization).

Author: Accepted the suggested change. For the remaining questions, I would prefer the user to go to the other page to learn more.

> ## Create a project

Author: I did consider that option before writing afresh in this topic. The problem is, the last step in the PL procedure has this step:

4. Pick a module and click Create New Pipeline to get started!

To create a project, perform the following steps:

> 

Reviewer 2: To create a project:

Author: Updated

1. In Harness, select **Service Reliability** > **Create Project**.
   ![Create Project](./static/change-impact-create-project-navigate.png)

> 2. In the About the Project page, do the following and then select **Save and Continue**: 

Author: Accepted

   - **Name**: Enter a name for the project.
   - **Color**: Select a color theme for the project.

>    - **Organization**: Select an organization for the project. If you don't yet have an organization in Harness, go to [Create a Harness Org](https://developer.harness.io/docs/platform/organizations-and-projects/create-an-organization#step-1-create-a-harness-org) to learn how to create an Organization.

Author: Accepted

> 3. In the Invite Collaborators (Optional) page, add team members to the project in the **Invite People to Collaborate** field.

Author: Accepted

4. In the **Role** field, assign a role to the collaborators.
5. Select **Add**.
6. Select **Save and Continue**.  
   Your project is created.

## Create a monitored service

> A Harness Monitored Service is a combination of a Harness service and environment. Service represents a microservice or other workloads, which means it is a logical entity that can be monitored. Environments represent your deployment targets such as QA, prod, and so on.

Author: Loved the suggestion! thank you!

> 

Reviewer 2: A Harness Monitored Service is a combination of a Harness service and environment. A service represents a microservice or other workloads, which means it is a logical entity that can be monitored. Environments represent your deployment targets such as QA, prod, and so on.

Author: Updated text:

A monitored service is a logical entity that combines a Harness service and environment. The service could be a microservice or any other workload that can be monitored. Environments represent your deployment targets such as QA, prod, and so on.

The following steps explain how to create a monitored service in Harness.

### Navigate to the "Create new monitored service" page

To create a monitored service:

> 1. In your Harness project, navigate to the **Service Reliability** module and then select **Monitored Services** > **+ New Monitored Service**.

Author: Updated

![Navigate to new monitored services page](./static/change-impact-moniterdservice-navigate.png)

> The Create new monitored service settings page appears.

Author: Agree. Changed to "he Create new monitored service settings page appears."

> ### Define monitored service

Author: Updated.

1. In the **Overview** section, enter the following information:
    * **Type**: Select **Application**.

>     * **Create or select a Service**: Select a Harness service that you want to monitor.

Author: Updated

>       To create a new service, perform the following steps:

Author: Updated

        
      1. Select **+ Add New**.  
        The New Service dialog appears.
      2. Enter the following information, and then select **Save**:
       

>           *  **Name** : Enter a name for the service. For example, _sample_service_.

Author: Updated

          *  **Description (Optional)**
          *  **Tag (Optional)** 

>     * **Create or Select an Environment**: Choose an appropriate environment.  

Author: Absolutely :)

  
        To create a new environment, perform the following steps:
    
      1. Select **+ Add New**. 
     
         The New Environment dialog appears.

>       2. Enter the following information and then select **Save**:

Author: Updated

            * **Name**: Enter a name for the environment.
            * **Description (Optional)**
            * **Tag (Optional)**
            * **Environment Type**: Select an environment type. The available options are **Production** and **Non-Production**.

>     Harness autopopulates the **Monitored Service Name** field by combining the **Service** and **Environment** names. This field is not editable.

Author: Accepted

> ### Add change source

Author: My mistake, can't blame frontend :)

> A Change Source monitors change events related to deployments, infrastructure changes, and incidents.  

Author: Accepted

> In the **Change Sources** section, you can either edit the existing change source or create a new one.

Author: Accepted

 
To create a new change source, do the following:

> 

Reviewer 2: To create a new change source:

Author: This looks correct but keeping the existing text based on comments from earlier reviews.

1. Select **+ Add New Change Source**.  
   The Add New Change Source dialog appears.

   ![Add new change source](./static/change-impact-moniterdservice-add-changesource.png)  

> 2. In the **Provider Type** dropdown, select the change source. Following are the available options:  

Author: Updated all the instances.

    - **Deployments**: You can choose **Harness CD NextGen** or **Harness CD**. 

>     - **Incident**: Select the incident management tool which monitors your application.

Author: Updated

3. In the **Source Name** field, enter a name for the change source.

> 4. In the **Application** dropdown, select an application ID. This field is available only if you have selected **Harness CD** as the deployment option.

Author: This looks straight forward - This field is available if you select Harness CD in **Provider Type**.

The new change source appears in the **Change Sources** list.

### Add a health source

> A Health Source monitors the health trends of the Monitored Service using logs or metrics collected from an APM or logging tool.

Author: - Corrected all instances of Health Source --> to health source
- Expanded APM

To add a new health source, do the following:  

> 

Reviewer 2: To add a new health source:

Author: This looks correct but keeping the existing text based on comments from earlier reviews.

1. Select **+ Add New Health Source**.  
   The Add New Health Source dialog appears.

   ![Add new health source](./static/change-impact-moniterdservice-add-healthsource.png)

2. In the **Select health source type** section, select the health source that monitors your service.
   
3. In the **Health Source Name** field, enter a name for the health source.
   
4. In the **Connect Health Source** section, do the following:

>    1. **Select Connector**: Select an existing connector. Connectors contain the information necessary to integrate and work with your health source.

Author: Updated

>    2. **Select Feature**: Select the feature associated with the health source. The available features depend on the health source that you select. For example, if you select **SumoLogic** as the health source, then the **SumoLogic Cloud Metrics** and **SumoLogic Cloud Logs** features are available. 

Author: Updated

5. Select **Next**.
   

> 6. In the **Customize Health Source** tab, the settings are displayed based on the health source you have selected in the **Define Health Source** tab.

Author: Changed to "On the **Customize Health Source** tab, configure the health source settings. The settings that are displayed depend on the health source that you select in the **Define Health Source** tab."

   
   For example, if you select AppDynamics as the **Health Source Type**, then the following settings are displayed:

>    - **Find an AppDynamics application**: Displays the list of applications associated with AppDynamics. Select the appropriated application.

Author: I would prefer "Select the application you want".

   
   - **Find an AppDynamics tier**: Displays a list of tiers associated with the selected application. Select the appropriate tier.
   
   - In the **Metrics Packs** section, select the metrics that you want to use for the health source. The options available are **Errors** and **Performance**.

    ![Health source metric settings](./static/change-impact-view-healthsource-config-metric.png)
   
7. Select **Submit**.  
   The new health source appears in the **Health Sources** list.
   

> 8.   On the top right-hand corner select **Save**.  

Reviewer 3: ```suggestion
8. In the upper-right corner select **Save**.  
```
(per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/upper-left-upper-right, though your wording is also used.)

Author: Noted, thank you!
